### PR TITLE
feat(scheduler): add csv support for get_file_metadata grpc method

### PR DESCRIPTION
**Pull request by etolbakov**
# Which issue does this PR close?

Closes #.
I was exploring the code and came across the `TODO` about the `csv` file format support. So decided to address it.

 # Rationale for this change
The functionality extension.

# What changes are included in this PR?
- add `csv` / `tlb` file format support
- fix minor typos

# Are there any user-facing changes?
No

### CODE REVIEW

File: ballista/scheduler/src/scheduler_server/grpc.rs [hunk 1]
```diff
@@ -297,13 +298,24 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> SchedulerGrpc
         // TODO shouldn't this take a ListingOption object as input?
 
         let GetFileMetadataParams { path, file_type } = request.into_inner();
-        let file_format: Arc<dyn FileFormat> = match file_type.as_str() {
+        let file_format: Result<Arc<dyn FileFormat>, Status> = match file_type.as_str() {
             "parquet" => Ok(Arc::new(ParquetFormat::default())),
```
Comment (etolbakov) [inline ballista/scheduler/src/scheduler_server/grpc.rs:302]: I've noticed that in the [benchmark](https://github.com/apache/datafusion-ballista/blob/04766d5fd7cbb15d583f736979c11de06c720928/benchmarks/src/bin/tpch.rs#L830) there's slightly different `default` for parquet:
```rust
ParquetFormat::default().with_enable_pruning(Some(true))
```
I wonder if we need consistency or it's ok to leave it as is?

File: ballista/scheduler/src/scheduler_server/grpc.rs [hunk 2]
```diff
@@ -297,13 +298,24 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> SchedulerGrpc
...
+            "tbl" => {
+                let format = CsvFormat::default()
+                    .with_delimiter(b'|')
+                    .with_has_header(false);
+                Ok(Arc::new(format))
+            }
             _ => Err(tonic::Status::unimplemented(
                 "get_file_metadata unsupported file type",
             )),
-        }?;
```
Comment (etolbakov) [inline ballista/scheduler/src/scheduler_server/grpc.rs:306]: I had to drop the "short circuit"  from here to resolve the rust compilation error:
```sh
Type mismatch [E0308] expected `Result<Arc<CsvFormat>, Status>`, but found `Result<Arc<ParquetFormat>, Status>` 
```
please let me know if there's a better way of doing things?

**Comment by etolbakov** [comment]
@andygrove 
Could you please take a look at this change when you have  time? 
Provided that there were no open ticket (but only todo in the code) I understand that maybe this change is of no use.

**Comment by etolbakov** [comment]
@andygrove just a ping in case this notification is lost.
Could you please take a look?

File: ballista/scheduler/src/scheduler_server/grpc.rs (hunk 1 above)
Comment (Dandandan) [reply ballista/scheduler/src/scheduler_server/grpc.rs:302]: I think pruning is default already, so in practice it will be the same.

**Comment by etolbakov** [comment]
Thank you for the feedback @Dandandan 
I'll resolve the conflicts and address your comments in the next couple of days!

**Comment by milenkovicm** [comment]
@Dandandan @etolbakov  `get_file_metadata` has been removed from latest code (#/1110). it was not clear what it was used for, no documentation nor any of the internal ballista code was using it. 

can you please elaborate what was the point of it ?

**Comment by etolbakov** [comment]
@milenkovicm @Dandandan 
sorry for the confusion, as per description I was exploring the code and came across the TODO about the csv file format support. So decided to address it.
If it doesn't seem to be useful enough or creates more hassle we can close the ticket. 
I'm happy to look into something else.

**Comment by milenkovicm** [comment]
As project was not actively maintained and lot of work accumulated we decide to change project's scope and remove code so maintainers have smaller code base to maintain ( some details in #1067 ). This specific functionality has been removed as there is no clear benefit of keeping it. 

Currently, there is active work ongoing towards #974

@etolbakov if you wish to contribute please join discord channel, there are ongoing discussions on tasks and priority

**Pull request closed without merging**